src/status_im/contexts/wallet/wallet_connect/utils/data_store.cljs

```diff
@@ -73,3 +73,25 @@
   (-> db
       get-db-current-request-event
       get-request-params))
+
+(defn get-network-from-request
+  [db]
+  (-> db
+      get-db-current-request-event
+      (get-in [:params :chainId])))
+
+(defn get-networks-from-proposal
+  "Find networks from current proposal as a string of comma separated values"
+  [db & [proposal]]
+  (let [current-proposal (get-in db [:wallet-connect/current-proposal :request])]
+    (cond
+      current-proposal
+      (->> (get-in db [:wallet-connect/current-proposal :session-networks])
+           vec
+           (string/join ","))
+
+      proposal
+      (->> (or (get-in proposal [:params :requiredNamespaces :eip155 :chains])
+               (get-in proposal [:requiredNamespaces :eip155 :chains]))
+           vec
+           (string/join ",")))))
```

Comment:
the networks you get from db's `:session-networks` and the ones you get from the proposal itself are a bit different, so we shouldn't mix them up when sending the metrics:

1. The session networks are the subset of the proposed networks, which both the dapp and our wallet support and with which **WILL** be used within the session. 
2. The proposal networks are all the networks that the dapp **CAN** establish a session with and are divided into required and optional.

IMO this should be two different functions, one for the proposal networks and another one for the session networks (if we need both of course), since the name is confusing and they don't share any steps.

Also, I don't think we get a lot of value from sending the proposal networks for successful approvals. We only need to send the "required" networks from the proposal when we don't support a required by the dApp (so we understand which networks that we don't yet support are in demand). When the approval is successful though, we should send the session networks only, since it includes both the required and the (supported) optional networks.